### PR TITLE
Fix: Extension inactive handling

### PR DESCRIPTION
**Pull request by amovar18**
## Description
Post moving to the service worker it was noticed that if there is no event received for 30 seconds the service worker is made to sleep. Due to making the service worker idle the service worker loses all the global data causing the 0 cookie issue. This PR aims to show a screen on DevTools for a user to reinitiate the syncCookieStore object and refresh the DevTools and the website.

## Relevant Technical Choices
- message passing from the service worker to PSAT.
- PSAT shows the screen with a button to refresh the panel and a message notifying the service worker was put to sleep.
- Once the user clicks on the button, a message is sent to the service worker to reinitialise the syncCookieStore object.
- Once done message is sent from the service worker to the PSAT to disable the message (This is to handle the multitab environment) and then the page is reloaded.

## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/a4fbac98-9a26-4f05-8a70-c23918b5caa7

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

### CODE REVIEW

File: packages/extension/src/serviceWorker/index.ts [hunk 1]
```diff
@@ -134,7 +146,19 @@ chrome.webRequest.onResponseStarted.addListener(
 chrome.webRequest.onBeforeSendHeaders.addListener(
   ({ url, requestHeaders, tabId, frameId, requestId }) => {
     (async () => {
-      const tabUrl = syncCookieStore?.getTabUrl(tabId) ?? '';
+      if (!syncCookieStore) {
```
Comment (ayushnirwal) [inline packages/extension/src/serviceWorker/index.ts:149]: Would you mind adding an explainer comment why `syncCookieStore` not being defined has the conclusion that the SW was in a sleep state?

**Review by ayushnirwal — APPROVED** [review]
LGTM. Other that minor code quality changes requested above.

**Comment by ayushnirwal** [comment]
@amovar18 Is this change still needed? I was not able to recreate the scenario which this PR aims to fix with a min of inactivity.

**Comment by mohdsayed** [comment]
Closing as its not required anymore

**Pull request closed without merging**